File: 백준/Silver/11057. 오르막 수/오르막 수.cc
#include <iostream>
#include <algorithm>
using namespace std;

int main() {
    int n;
    cin >> n;
    
    int dp[1001][10] = {0};
    int sum=0;
    for(int i=0;i<10;i++) dp[1][i] = 1;
    for(int i=1;i<=n;i++)
    {
        for(int j=0;j<10;j++)
        {
            
            for(int x=j;x<10;x++) dp[i][j] = (dp[i][j]+dp[i-1][x])%10007;
        }
    }
    for(int i=0;i<10;i++) sum = (sum+dp[n][i])%10007;
    cout << sum;
}

File: 백준/Silver/11057. 오르막 수/오르막 수.py
n = int(input())

d = [[0 for i in range(10)] for j in range(1001)]

for i in range(10):
    d[0][i] = 1
    
for i in range(1,n+1):
    for j in range(10):
        d[i][j] = sum(d[i-1][j:])
    
print(sum(d[n-1])%10007)